src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncMethodInAsyncContext.cs

```diff
@@ -0,0 +1,282 @@
+                }
+
+                context.RegisterOperationAction(context =>
+                {
+                    if (IsInTaskReturningMethodOrDelegate(context, syncBlockingTypes))
+                    {
+                        if (context.Operation is IInvocationOperation invocationOperation)
+                        {
+                            if (InspectAndReportBlockingMemberAccess(context, syncBlockingSymbols, SymbolKind.Method))
+                            {
+                                // Don't return double-diagnostics.
+                                return;
+                            }
+
+                            // Also consider all method calls to check for Async-suffixed alternatives.
+                            var semanticModel = context.Operation.SemanticModel;
+                            SymbolInfo symbolInfo = semanticModel.GetSymbolInfo(context.Operation.Syntax, context.CancellationToken);
+
+                            if (symbolInfo.Symbol is IMethodSymbol methodSymbol &&
+                                !methodSymbol.Name.EndsWith(MandatoryAsyncSuffix, StringComparison.Ordinal) &&
+                                !HasAsyncCompatibleReturnType(methodSymbol, syncBlockingTypes))
+                            {
+                                string asyncMethodName = methodSymbol.Name + MandatoryAsyncSuffix;
+                                IEnumerable<IMethodSymbol> methodSymbols = semanticModel.LookupSymbols(
+                                    context.Operation.Syntax.GetLocation().SourceSpan.Start,
+                                    methodSymbol.ContainingType,
+                                    asyncMethodName,
+                                    includeReducedExtensionMethods: true)
+                                    .OfType<IMethodSymbol>();
+
+                                string containingMethodName = "";
+                                if (context.ContainingSymbol is IMethodSymbol parentMethod)
+                                {
+                                    containingMethodName = parentMethod.Name;
+                                }
+
+                                SyntaxNode invokedMethodName = context.Operation.Syntax;
+
+                                foreach (IMethodSymbol method in methodSymbols)
+                                {
+                                    if (!method.HasAttribute(systemObsoleteAttribute)
+                                        && HasSupersetOfParameterTypes(method, methodSymbol)
+                                        && method.Name != containingMethodName
+                                        && HasAsyncCompatibleReturnType(method, syncBlockingTypes))
+                                    {
+                                        Diagnostic diagnostic = invocationOperation.CreateDiagnostic(
+                                            Descriptor,
+                                            invokedMethodName.ToString(),
```

Comment:
This produces a string from SyntaxNode of the `IInvocationOperation`, which seems to be including the arguments. Consider confirming the message produce by a test, and see if the following suggestion fixes it:

```suggestion
                                            invocationOperation,TargetMethod.Name,
```

Reply:
That also doesn't look that good, I am updating to use `Symbol.ToDisplayString()` instead